xclim/indices/stats.py

```diff
@@ -585,20 +587,204 @@ def dist_method(
+        Indexing parameters to compute the indicator on a temporal subset of the data.
+        It accepts the same arguments as :py:func:`xclim.indices.generic.select_time`.
+
+    Returns
+    -------
+    xarray.DataArray
+        Standardized Index fitting parameters. The time dimension of the initial array is reduced to
+
+    Notes
+    -----
+    Supported combinations of `dist` and `method` are:
+    * Gamma ("gamma") : "ML", "APP", "PWM"
+    * Log-logistic ("fisk") : "ML", "APP"
+    """
+    # "WPM" method doesn't seem to work for gamma or pearson3
+    dist_and_methods = {"gamma": ["ML", "APP", "PWM"], "fisk": ["ML", "APP"]}
+    if dist not in dist_and_methods:
+        raise NotImplementedError(f"The distribution `{dist}` is not supported.")
+    if method not in dist_and_methods[dist]:
+        raise NotImplementedError(
+            f"The method `{method}` is not supported for distribution `{dist}`."
+        )
+
+    if offset is not None:
+        with xr.set_options(keep_attrs=True):
+            da = da + convert_units_to(offset, da, context="hydro")
+
+    da, group = preprocess_standardized_index(da, freq, window, **indexer)
+    params = da.groupby(group).map(fit, dist=dist, method=method)
+    cal_range = (
+        da.time.min().dt.strftime("%Y-%m-%d").item(),
+        da.time.max().dt.strftime("%Y-%m-%d").item(),
+    )
+    params.attrs = {
+        "calibration_period": cal_range,
+        "freq": freq,
+        "window": window,
+        "scipy_dist": dist,
+        "method": method,
+        "group": group,
+        "time_indexer": indexer,
+        "units": "",
+        "offset": offset,
+    }
+    return params
+
+
+def standardized_index(da: xr.DataArray, params: xr.DataArray):
+    """Compute standardized index for given fit parameters.
+
+    This computes standardized indices which measure the deviation of  variables in the dataset compared
+    to a reference distribution. The reference is a statistical distribution computed with fitting parameters `params`
+    over a given calibration period of the dataset. Those fitting parameters are obtained with
+    ``xclim.standardized_index_fit_params``.
+
+    Parameters
+    ----------
+    da : xarray.DataArray
+        Input array.
+        Resampling and rolling operations should have already been applied using
+        ``xclim.indices.preprocess_standardized_index``.
+    params : xarray.DataArray
+        Fit parameters computed using ``xclim.indices.stats.standardized_index_fit_params``.
+    """
+    group = params.attrs["group"]
+
+    def reindex_time(da, da_ref):
+        if group == "time.dayofyear":
+            da = da.rename(day="time").reindex(time=da_ref.time.dt.dayofyear)
+            da["time"] = da_ref.time
+            da = resample_doy(da, da_ref)
+        elif group == "time.month":
+            da = da.rename(month="time").reindex(time=da_ref.time.dt.month)
+            da["time"] = da_ref.time
+        return da.chunk({"time": -1})
```

Comment:
Hum, are we rechunking even if dask was never used ?

```suggestion
        return da if not uses_dask(da) else da.chunk({"time": -1})
```

Reply:
oups, I had those checks earlier, must have dropped them by accident